tests/components/fronius/__init__.py

```diff
@@ -32,55 +36,77 @@ async def setup_fronius_integration(
     return entry
 
 
+def _load_and_patch_fixture(
+    patch_filename: str, patch_values: list[tuple[list[str], Any]]
+) -> Callable[[str, str | None], str]:
+    """Return a fixture loader that patches values at nested keys for a given filename."""
+
+    def load_and_patch(filename: str, integration: str):
+        """Load a fixture and patch given values."""
+        text = load_fixture(filename, integration)
+        if patch_filename != filename.split("/")[-1]:
+            return text
+        _loaded = json.loads(text)
```

Comment:
You should return it unchanged if no patching is done. Running ky through loads/dumps is both slow and risk changing the fixture data in unplanned ways.

Reply:
It is, isn't it? 
Line 49 is only invoked when patching. When not patching, the `load_fixture` function result is used directly (Line 48).

Reply:
Oh its that weird split i didnt understand.

A dict of fixes with the filename as key would make sense. Also i suggest you add the full filename. So the check is just:

```
if not (patch_values := patches.get(filename)):
  return text
```

The prefix to filename seem know at call site.

Reply:
A dict seems like a good idea!